### DISCUSSION TITLE
Support alternate share hostname

### DISCUSSION THREAD

**Discussion by samcday** [navidrome/navidrome#3117] [Ideas]
I'm self-hosting Navidrome on my home lab (and have been happily doing so for many months now, thanks for a wonderful project!), which I access either directly via internal (RFC1918) network when I'm physically at home, or via my Tailscale tailnet when I'm out and about.

I want to be able to also share Navidrome links with folks that don't have access to my internal home network/tailnet. I have the ability to make stuff available publicly (personally, I use Cloudflare Tunnels), so I'd like to be able to have the share URLs generated with a different hostname that I can ensure is publicly available.

TL;DR, if I'm hosting navidrome on `navidrome.samcday.com`, I want to be able to run Navidrome with `ND_SHAREHOST=navidrome-share.samcday.com` and thus have share URLs generated as `https://navidrome-share.samcday.com/share/...`.

**Comment by deluan** [comment]
I think adding a config option for this is kind of overkill, as you have two ways to accomplish what you want:

1) You can change the address to whatever host you are using, just keep the `/share/XXXX` part. The new URL should work the same way. Ex: http://navidrome.samcday.com/share/BcrKoQDBBY  =>  https://navidrome-share.samcday.com/share/BcrKoQDBBY



2) Login to your Navidrome using the https://navidrome-share.samcday.com address, go to the "Share" menu and the links in the list will have this address as host.

**Reply by samcday** [reply]
In both cases, it requires me to do some manual work *every time* I want to share a link, though.

My Subsonic clients are configured to point to `https://navidrome.samcday.com`, because this is the optimal path for traffic to take for *me personally*. This hostname resolves to an RFC1918 address. When I am at home, my clients are connecting directly to Navidrome via my home network. When I am outside of the house/home-network, my VPN handles routing that private `10.0.1.x` address.


`navidrome-public.samcday.com` would be *always public*, to guarantee that people not on my home network or connected to my private VPN can access the shares I sent.

With what you've proposed, it means I can never share URLs from my Subsonic clients (because they should *always* connect to `navidrome.samcday.com`).

**Reply by deluan** [reply]
Sorry for the late response, but just to let you know that we will be adding a new config to set the share host, as you suggested. The new release will be out this weekend. Thanks for the patience :)

**Reply by deluan** [reply]
Implemented in https://github.com/navidrome/navidrome/pull/3265

**Reply by samcday** [reply]
Wonderful news! I just upgraded to `0.53.1` and confirmed the new ShareURL configuration is exactly what I was hoping for. Thanks a bunch @caiocotts :) And huge thanks to you @deluan (and all the other contributors, ofc!) for Navidrome :heart_eyes_cat:

**Reply by metalheim** [reply]
@samcday  can you mark this discussion as "Answered" please? Keeps the Discussions section tidy :)

**Comment by samcday** [comment]
Resolving this discussion as the requested feature was implemented in #3265 and released in version `0.53.0`
